WordPress/src/main/java/org/wordpress/android/ui/plans/PlansActivity.java

```diff
@@ -367,24 +370,46 @@ private void startPurchaseProcess(int position) {
+                            }
+
+                            AnalyticsUtils.trackWithCurrentBlogDetails(
+                                    result.isSuccess() ? AnalyticsTracker.Stat.PRODUCT_PURCHASED : AnalyticsTracker.Stat.PRODUCT_PAYMENT_ERROR,
+                                    analyticsProperties
+                            );
+
                             if (result.isSuccess()) {
-                                if (info != null) {
-                                    /*
-                                        Sync the purchase info with the wpcom backend, and enabled the product on the site.
-                                        We need to use an app setting here for security reasons.
-                                        If something bad happens during this sync, we need to re-sync it later (See onAppComesFromBackground in WordPress.java)
-                                        Without this initial sync the backend doesn't have any info about the purchase, and the product will NOT be enabled
-                                        without a manual action on backend side.
-                                     */
-                                    AppPrefs.setInAppPurchaseRefreshRequired(true);
-                                    PlansUtils.synchIAPsWordPressCom();
-                                    AppLog.d(AppLog.T.PLANS, "Purchase: " + info.toString());
-                                    AppLog.d(AppLog.T.PLANS, "You have bought the " + info.getSku() + ". Excellent choice, adventurer!");
-                                    boolean isBusinessPlan = (mViewPager.getCurrentItem() == mViewPager.getAdapter().getCount() - 1);
-                                    Intent intent = new Intent(PlansActivity.this, PlanPostPurchaseActivity.class);
-                                    intent.putExtra(PlanPostPurchaseActivity.ARG_IS_BUSINESS_PLAN, isBusinessPlan);
-                                    startActivity(intent);
-                                }
+                                /*
+                                    Sync the purchase info with the wpcom backend, and enabled the product on the site.
+                                    We need to use an app setting here for security reasons.
+                                    If something bad happens during this sync, we need to re-sync it later (See onAppComesFromBackground in WordPress.java)
+                                    Without this initial sync the backend doesn't have any info about the purchase, and the product will NOT be enabled
+                                    without a manual action on backend side.
+                                 */
+                                AppPrefs.setInAppPurchaseRefreshRequired(true);
+                                PlansUtils.synchIAPsWordPressCom();
+                                boolean isBusinessPlan = (mViewPager.getCurrentItem() == mViewPager.getAdapter().getCount() - 1);
```

Comment:
I should've noticed this on an earlier PR, but this seems an unreliable way to determine whether the purchased plan is a business plan.